fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -95,8 +96,9 @@ public AggregationAnalyzer(ConnectContext session, AnalyzeState analyzeState, Li
         this.orderByScope = orderByScope;
         this.analyzeState = analyzeState;
         this.groupingExpressions = groupingExpressions;
+        // We need to filter out subfields of struct type SlotRef here.
         this.groupingFields = groupingExpressions.stream()
-                .filter(analyzeState.getColumnReferences()::containsKey)
+                .filter(filterSubfieldsFromColumnReference(analyzeState.getColumnReferences())::containsKey)
                 .map(analyzeState.getColumnReferences()::get)
```

Comment:
Think about a case:
```
## c1 col def is struct<a int, b int, c struct<a int, b int>>
select c1.c.a, c1.c.b from tbl group by c1.c; 
```
I think it's a valid sql. Filter the subFieldCol may fail to analyze this sql.